### PR TITLE
Allow merge requests on Views with forks (SYN-7401)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -4496,6 +4496,81 @@ async def _addView(self, vdef):
...
+            self.dynitems.pop(layriden)
+
+            await self.hive.pop(('cortex', 'layers', layriden))
+
+            await layr.delete()
+
+            layr.deloffs = nexsitem[0]
+
+        for cview in self.views.values():
+            if layr in cview.layers:
```
Comment (MichaelSquires) [inline synapse/cortex.py:4570]: It seems like we should only be deleting this layer from other views if the other view is a child of the current view (the only being deleted)? What if I had manually constructed a view with this layer in it? It just disappears from under me? What if it was my current write layer?

File: synapse/cortex.py [hunk 2]
```diff
@@ -4496,6 +4496,81 @@ async def _addView(self, vdef):
...
+            await self.auth.delAuthGate(layriden)
+            self.dynitems.pop(layriden)
+
+            await self.hive.pop(('cortex', 'layers', layriden))
+
+            await layr.delete()
+
+            layr.deloffs = nexsitem[0]
+
+        for cview in self.views.values():
```
Comment (MichaelSquires) [inline synapse/cortex.py:4569]: Can we move the layer loop above the first view loop and then combine this view loop with the other view loop?

File: synapse/cortex.py (hunk 1 above)
Comment (Cisphyx) [reply synapse/cortex.py:4570]: Yeah I suppose we should leave it alone if it is used outside the current stack.

File: synapse/cortex.py (hunk 2 above)
Comment (Cisphyx) [reply synapse/cortex.py:4569]: Yeah i'll rearrange this

File: synapse/cortex.py [hunk 3]
```diff
@@ -4496,6 +4496,84 @@
...
+        '''
+        view = self.views.get(iden)
+        if view is None:
+            raise s_exc.NoSuchView(mesg=f'No such view {iden=}', iden=iden)
+
+        if view.info.get('protected'):
+            mesg = f'Cannot delete view ({iden}) that has protected set.'
+            raise s_exc.CantDelView(mesg=mesg)
+
+        layriden = view.layers[0].iden
```
Comment (mikemoritz) [inline synapse/cortex.py:4519]: minor nit, but this is unused

Comment (Cisphyx) [reply synapse/cortex.py:4519]: ah, yeah i added it and forgot to use it, updated

**Review by mikemoritz — APPROVED** [review]

**Pull request merged**